src/api/rasters.ts

```diff
@@ -0,0 +1,290 @@
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({"source": null})
+  });
+
+  // Try to delete source, ignore errors
+  await fetch(raster.raster_sources[0], {
+    credentials: "same-origin",
+    method: "DELETE",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({})
+  });
+
+  // Create new raster source from old one
+  const rasterSourceBody = {
+    name: oldSource.name,
+    description: oldSource.description,
+    organisation: oldSource.organisation,
+    access_modifier: "Private", // Always
+    supplier: oldSource.supplier,
+    supplier_code: oldSource.supplier_code,
+    temporal: oldSource.temporal,
+    interval: oldSource.temporal ? oldSource.interval : undefined
+  };
+
+  const newSourceResponse = await fetch('/api/v4/rastersources/', {
+    credentials: "same-origin",
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(rasterSourceBody)
+  });
+  const newSource = await newSourceResponse.json();
+
+  // Patch layer so it has the new one
+  return fetch(`/api/v4/rasters/${uuid}/`, {
+    credentials: "same-origin",
+    method: "PATCH",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({source: {
+      graph: {
+        raster: [
+          "lizard_nxt.blocks.LizardRasterSource",
+          newSource.uuid
+        ]
+      },
+      name: "raster"
+    }})
+  });
+};
+
+export const flushRasters = (uuids: string[]) => {
+  return Promise.all(uuids.map(flushRaster));
+}
+
+export const deleteRaster = async (uuid: string) => {
+  const deleteOpts: RequestInit = {
+    credentials: "same-origin",
+    method: "DELETE",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({})
+  };
+
+  // Re-fetch raster so we have up to date information here
+  const raster = await fetchRasterV4(uuid);
+
+  // Send a delete request to delete the raster
+  await fetch(`/api/v4/rasters/${uuid}/`, deleteOpts);
+
+  // If this isn't a Geoblock, also delete the first raster source.
+  if (!raster.is_geoblock && raster.raster_sources) {
+    const source = raster.raster_sources[0];
+    if (source) {
+      // This can fail for various reasons but we ignore all of them:
+      // 412 "Precondition failed" -- another layer is using the source
+      // 403 "Permission denied" -- user doesn't have write rights to the source
+      // 404 -- user doesn't have read rights to the source
+      // On success, 204 Gone is returned.
+      await fetch(source, deleteOpts);
+    }
+  }
+
+  return raster;
+};
+
+export const deleteRasters = (uuids: string[]) => {
+  return Promise.all(uuids.map(deleteRaster));
+}
+
+export const listTemporalRastersContainingV3 = async (searchString: string) => {
+  const response = await fetch(
+    // show all temporal rasters the user has access to
+    `/api/v3/rasters/?page_size=0&name__icontains=${searchString}&first_value_timestamp__isnull=false`,
```

Comment:
I didn't change this, just moved it to this file because I wanted to have all raster-related API calls in the same place, not within React components.

It is used when created an alarm on (a location on) a temporal raster, so there only temporal rasters make sense and we do need a way to filter for them, yes.

Reply:
I created a JIRA ticket to solve this. If you replace the filter on `first_value_timestamp` with `temporal=true`, it will start to work as soon as the backend is up to speed.

Reply:
> Non-temporal rasters also have a first_value_timestamp. I don't see a way to filter for temporal rasters currently. Didn't know it was necessary?
> 
> Is it necessary? --> then we need a fix on the backend

@caspervdw 
@remcogerlich 

I see in the current version no usage of this feature, 
but in the future it would be useful if a user can search on it. 
If you agree we can make a ticket for it.

Reply:
@Derryrover as I said it is used, for the choice of raster for raster-intersection alarms.

@caspervdw that change will be in v4 only? Are there differences in fields returned between v3 and v4?

Reply:
Change is done in v3 and v4 in https://github.com/nens/lizard-nxt/pull/3597

Reply:
Top